Tools/Scripts/libraries/webkitcorepy/webkitcorepy/credentials.py

```diff
@@ -53,6 +61,8 @@ def credentials(url, required=True, name=None, prompt=None, key_name='password',
     key_prompted = False
 
     for attempt in range(retry):
+        if not attempt and ignore_entry:
```

Comment:
what is this if statement for?

Reply:
We have a loop here so that if a user enters their password incorrectly, we re-prompt them before failing. However, the first iteration of the loop will retrieve the username and password from the keychain. If we've asked to ignore the entry in keychain, we want to skip that first iteration.